packages/frontend/src/features/sqlRunner/hooks/useTables.tsx

```diff
@@ -1,25 +1,79 @@
 import { type ApiError, type ApiWarehouseCatalog } from '@lightdash/common';
 import { useQuery } from '@tanstack/react-query';
+import Fuse from 'fuse.js';
 import { lightdashApi } from '../../../api';
 
 export type GetTablesParams = {
     projectUuid: string;
+    search: string | undefined;
 };
 
-const fetchTables = async ({ projectUuid }: GetTablesParams) =>
+const fetchTables = async ({
+    projectUuid,
+}: Pick<GetTablesParams, 'projectUuid'>) =>
     lightdashApi<ApiWarehouseCatalog>({
         url: `/projects/${projectUuid}/sqlRunner/tables`,
         method: 'GET',
         body: undefined,
     });
 
-export const useTables = ({ projectUuid }: GetTablesParams) => {
-    return useQuery<ApiWarehouseCatalog, ApiError>({
-        queryKey: ['sqlRunner', 'tables', projectUuid],
+type TablesBySchema =
+    | {
+          schema: keyof ApiWarehouseCatalog['results'][0];
+          tables: ApiWarehouseCatalog['results'][0]['tables'];
+      }[]
+    | undefined;
+
+export const useTables = ({ projectUuid, search }: GetTablesParams) => {
+    return useQuery<ApiWarehouseCatalog, ApiError, TablesBySchema>({
+        queryKey: ['sqlRunner', 'tables', projectUuid, search],
```

Comment:
Having the `search` as part of the query key is causing a new request and the results to flash on each search. When the denounced search is applied the state of the query changes to `loading` showing "no results" 

<img width="235" alt="Screenshot 2024-07-03 at 10 41 32" src="https://github.com/lightdash/lightdash/assets/9117144/58d55ce5-35e1-4dd5-8b79-d65de3d2db7c">

Reply:
ty, will address these 2 cases 👍

Reply:
Removed `search` from `queryKey`: https://github.com/lightdash/lightdash/pull/10586/commits/8311d3f897b7921b6ad36e857d80b05e8899547b

Only show `no results` if fetching is done **and** there's no `data`: https://github.com/lightdash/lightdash/pull/10586/commits/7149130d5504687c6e3b87a9fa4c3496f82c6ace